format/mp4/boxes.go

```diff
@@ -1367,27 +1367,48 @@ func decodeBox(ctx *decodeContext, d *decode.D, typ string) {
 	case "sgpd":
 		version := d.FieldU8("version")
 		d.FieldU24("flags")
-		d.FieldUTF8("grouping_type", 4)
+		var groupingType = d.FieldUTF8("grouping_type", 4)
 		var defaultLength uint64
 		if version == 1 {
 			defaultLength = d.FieldU32("default_length")
-		}
-		if version >= 2 {
+		} else if version >= 2 {
 			d.FieldU32("default_sample_description_index")
 		}
 		entryCount := d.FieldU32("entry_count")
-		d.FieldArray("entries", func(d *decode.D) {
-			for i := uint64(0); i < entryCount; i++ {
-				entryLen := defaultLength
-				if version == 1 {
-					if defaultLength == 0 {
-						entryLen = d.FieldU32("description_length")
-					} else if entryLen == 0 {
-						d.Fatalf("sgpd groups entry len <= 0 version 1")
-					}
+
+		d.FieldStructNArray("entries", groupingType+" entry", int64(entryCount), func(d *decode.D) {
```

Comment:
Most other boxes uses just entries/entry which i think is fine, don't think we need to repeat the type name. btw convention for the names are snake_case.

Reply:
The issue was that there are two different box types depending on `groupingType`. The intent was to indicate it somehow, but I'll just do the common way.